autodist/autodist.py

```diff
@@ -119,12 +129,22 @@ def _compile_strategy(self, strategy):
 
     def _setup(self, strategy):
         """Prepare for the execution."""
-        if IS_AUTODIST_CHIEF:
-            # we should only have one single coordinator for one single AutoDist() instance scope,
-            # even though we could have multiple strategies.
-            self._coordinator = Coordinator(strategy=strategy, cluster=self._cluster)
-            self._cluster.start()
-            self._coordinator.launch_clients()
+        if not IS_ADAPTDL:
+            if IS_AUTODIST_CHIEF:
+                # we should only have one single coordinator for one single AutoDist() instance scope,
+                # even though we could have multiple strategies.
+                self._coordinator = Coordinator(strategy=strategy, cluster=self._cluster)
+                self._cluster.start()
+                self._coordinator.launch_clients()
+        else:
+            if IS_AUTODIST_CHIEF:
+                self._coordinator = Coordinator(strategy=strategy, cluster=self._cluster)
```

Comment:
Would it be better if we create different Coordinator classes based on the cluster mode?

Reply:
Good suggestion. I tried similar format like you suggest. But I think the current version is more readable in autodist.py though more lengthy. Its easy to maintain this way since autodist.py is the first file to look at.